tslib.es6.js

```diff
@@ -350,17 +350,17 @@ export function __disposeResources(env) {
 
 export function __rewriteRelativeImportExtension(path, preserveJsx) {
     if (typeof path === "string" && path[0] === "." && (path[1] === "/" || path[1] === "." && path[2] === "/")) {
-        if (path.substring(path.length - 4) === ".tsx") {
+        if (path.substring(path.length - 4).toLowerCase() === ".tsx") {
             return path.substring(0, path.length - 4) + (preserveJsx ? ".jsx" : ".js");
         }
-        if (path.substring(path.length - 3) === ".ts") {
-            var dot = path.lastIndexOf(".", path.length - 4);
-            if (dot >= 0 && (path.substring(dot - 2, dot) === ".d" || path.substring(dot, dot + 2) === ".d")) {
-                return path;
+        var extMatch = path.match(/\.[cm]?ts$/i);

```

Comment:
You could possibly have handled the lookbehind as a regular capture group, e.g.:

```js
var extMatch = path.match(/(\.d)?\.([cm]?)ts$/i);
if (extMatch && !extMatch[1]) {
    return path.substring(0, extMatch.index + 1) + extMatch[2].toLowerCase() + "js";
}
```

or as the original call to `replace`:

```js
return path.replace(/(\.d)?\.([cm]?)ts$/i, function (m, d, ext) {
    return d ? m : "." + ext.toLowerCase() + "js";
});
```

Reply:
Given your comment below, a more concise implementation might be:

```js
if (typeof path === "string" && /^\.\.?\//.test(path)) {
    return path.replace(/\.(tsx)$|(\.d(?:\.\w+(?!$))?)?(\.[cm]?)ts$/i, function (m, tsx, d, ext) {
        return tsx ? preserveJsx ? ".jsx" : ".js" :
            d ? m : ext.toLowerCase() + "js";
    });
}
return path;
```
which checks first for a trailing `.tsx` and if that fails then a trailing `.ts`/`.cts`/`.mts` so long as it isn't prefixed by `.d` or `.d.whatever`.

Reply:
Combined the tsx case into my own regex that passes the current test suite